### PR TITLE
[VariantBundle] Add possibility to exclude Variants from AttributeCollector via AttributePreconditionEvent

### PR DESCRIPTION
**Pull request by almshadi**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no

### CODE REVIEW

File: src/CoreShop/Component/Variant/AttributeCollector.php [hunk 1]
```diff
@@ -72,7 +79,11 @@ public function getAttributes(array $products, bool $showInList = false): array
                     $group->setGroup($attributeGroup);
                     $group->setType(get_class($attribute));
 
-                    $resolvedGroups[$attributeGroup->getId()] = $group;
+                    $this->eventDispatcher->dispatch($event = new variantAvailabilityEvent($product), 'coreshop.attribute.collector.preCondition');
```
Comment (jdreesen) [inline src/CoreShop/Component/Variant/AttributeCollector.php]:
```suggestion
                    $event = $this->eventDispatcher->dispatch(new VariantAvailabilityEvent($product), 'coreshop.attribute.collector.preCondition');
```

File: src/CoreShop/Component/Variant/Event/variantAvailabilityEvent.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * @license    https://www.coreshop.org/license     GPLv3 and CCL
+ *
+ */
+
+namespace CoreShop\Component\Variant\Event;
+
+use CoreShop\Component\Core\Model\ProductInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class variantAvailabilityEvent extends Event
```
Comment (jdreesen) [inline src/CoreShop/Component/Variant/Event/variantAvailabilityEvent.php]:
```suggestion
class VariantAvailabilityEvent extends Event
```

The file also needs to be renamed.

File: src/CoreShop/Component/Variant/Event/variantAvailabilityEvent.php
Comment (jdreesen) [inline src/CoreShop/Component/Variant/Event/variantAvailabilityEvent.php:1]: The file name still starts with a lower case letter.

Comment (dpfaffenbauer) [reply src/CoreShop/Component/Variant/Event/variantAvailabilityEvent.php:1]: @almshadi please fix that

Comment (almshadi) [reply src/CoreShop/Component/Variant/Event/variantAvailabilityEvent.php:1]: done

**Review by dpfaffenbauer — APPROVED** [review]

File: src/CoreShop/Component/Variant/Event/VariantAvailabilityEvent.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+namespace CoreShop\Component\Variant\Event;
+
+use CoreShop\Component\Core\Model\ProductInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class VariantAvailabilityEvent extends Event
+{
+    private bool $conditionMet = true;
+
+    public function __construct(private ProductInterface $product)
```
Comment (dpfaffenbauer) [inline src/CoreShop/Component/Variant/Event/VariantAvailabilityEvent.php]: @almshadi Use VariantAwareInterface here

**Review by dpfaffenbauer — CHANGES_REQUESTED** [review]

File: src/CoreShop/Component/Variant/Event/VariantAvailabilityEvent.php (hunk 3 above)
Comment (almshadi) [reply src/CoreShop/Component/Variant/Event/VariantAvailabilityEvent.php]: fixed

**Review by dpfaffenbauer — APPROVED** [review]

**Comment by dpfaffenbauer** [comment]
🎉

**Pull request merged**